addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2822,6 +2822,220 @@ options.registry.ReplaceImage = options.Class.extend({
     },
 });
 
+options.registry.ConditionalVisibility = options.Class.extend({
+    optionsAttributes: [
+        {
+            selectName: 'visibilityValueCountry',
+            dataName: 'data-country',
+        },
+        {
+            selectName: 'visibilityValueLang',
+            dataName: 'lang',
+        },
+        {
+            selectName: 'visibilityValueUtmMedium',
+            dataName: 'data-utm-medium',
+        },
+        {
+            selectName: 'visibilityValueUtmSource',
+            dataName: 'data-utm-source',
+        },
+        {
+            selectName: 'visibilityValueUtmCampaign',
+            dataName: 'data-utm-campaign',
+        },
+    ],
+    /**
+     * @constructor
+     */
+    init() {
+        this._super(...arguments);
+        if (this.$target[0].dataset.visibility === 'conditional') {
+            // hack to use the same visibility pannel as the popups
+            this.$target[0].classList.add('o_snippet_invisible');
+            this.trigger_up('snippet_option_visibility_update', { show: true });
+        }
+    },
+    /**
+     * Inserts or deletes record's id and value in target's data-attributes
+     * if no ids are selected, deletes the attribute
+     *
+     * @see this.selectClass for parameters
+     */
+    selectRecord(previewMode, widgetValue, params) {
```

Comment:
Check the organisation of options methods. There should be a "options" section comment before options and "lifecycle" method should be place before (onFocus, cleanForSave, etc).